src/run_releasenotes_check.py

```diff
@@ -0,0 +1,94 @@
+gitIssueNumber = os.getenv('GIT_ISSUE_NUMBER')
+# Boolen if set to null, just prints on the console.
+addComment = os.getenv('ADD_COMMENT', default="true")
+# Token used to add comment on an issue
+userToken = os.getenv('GITHUB_TOKEN')
+
+
+def main(gitLogDate: str, addComment: str) -> int:
+    value_matrix = []  # type: List[Any]
+    args = CheckoutArgs()
+    console.configure(level=args.logging_level)
+    manifest = InputManifest.from_file(args.manifest)
+    writer = MarkdownTableWriter(
+        table_name=f"{manifest.build.name} Commits(after {gitLogDate}) & Release Notes info",
+        headers=["Repo", "Branch", "CommitID", "Release Notes"],
+        value_matrix=value_matrix
+    )
+    with TemporaryDirectory(keep=True, chdir=True) as work_dir:
+        logging.info(f"Checking out into {work_dir.name}")
+        for component in manifest.components.select():
+            logging.info(f"Checking out {component.name}")
+            repo_list = []  # type: List[Any]
+            if type(component) is InputComponentFromSource:
+                with GitRepository(
+                    component.repository,
+                    component.ref,
+                    os.path.join(work_dir.name, component.name),
+                    component.working_directory,
+                ) as repo:
+                    logging.debug(f"Checked out {component.name} into {repo.dir}")
+                    gitLogCmd = f'git -C {repo.dir} log --after={gitLogDate} --pretty=format:"%h"'
+                    gitLog = subprocess.check_output(gitLogCmd.split()).decode()
+                    repo_list.append(component.name)
+                    repo_list.append(component.ref)
+                    repo_list.append(gitLog)
+                    release_notes = "NO"
+                    if component.name == 'OpenSearch' or component.name == 'OpenSearch-Dashboards':
+                        release_notes_check = f'.release-notes-{manifest.build.version}.md'
+                    else:
+                        release_notes_check = f'.release-notes-{manifest.build.version}.0.md'
+                    try:
+                        if any(fname.endswith(release_notes_check) for fname in os.listdir(f'{repo.dir}/release-notes/')):
+                            release_notes = "YES"
+                    except FileNotFoundError:
+                        print("No such release-notes file or directory")
+                        release_notes = "NULL"
+                    repo_list.append(release_notes)
+                    value_matrix.append(repo_list)
+    writer.write_table()
+    writer.dump("table.txt")
+    if addComment == "true":
+        if os.environ.get('GIT_ISSUE_NUMBER') is not None:
+            comment_under_issue(gitIssueNumber, 'table.txt', userToken)
+        else:
+            print("No GIT_ISSUE_NUMBER environmental value passed")
+            exit()
+    subprocess_cmd('rm table.txt')
+    return 0
+
```

Comment:
All these can be put under the temp dir so the cleanup is automatic after completion.

Reply:
Good point, just pushed in my latest commit @peterzhuamazon.